### PR TITLE
chore: bump pymssql version

### PR DESCRIPTION
**Pull request by gnought**
### SUMMARY
pymssql >=2.2.8 is required to compatible with cython v3 https://github.com/pymssql/pymssql/pull/803

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -173,7 +173,7 @@ def get_git_sha() -> str:
         "impala": ["impyla>0.16.2, <0.17"],
         "kusto": ["sqlalchemy-kusto>=2.0.0, <3"],
         "kylin": ["kylinpy>=2.8.1, <2.9"],
-        "mssql": ["pymssql>=2.1.4, <2.2"],
+        "mssql": ["pymssql>=2.2.8"],
```
Comment (john-bodley) [inline setup.py:176]: Would you mind re-including an major version upper bound to prevent potential regressions due to major version bumps?

**Review by john-bodley — COMMENTED** [review]
Thanks @gnought for the PR. I've left an inline comment, but could you also update the PR title/description to ensure CI passes.

File: setup.py (hunk 1 above)
Comment (gnought) [reply setup.py]: @john-bodley thx for your advise, fixed.

**Comment by gnought** [comment]
@john-bodley fixed, all checks are passed.

**Review by john-bodley — APPROVED** [review]

**Pull request merged**